server/aesdsocket.c

```diff
@@ -0,0 +1,344 @@
+		syslog(LOG_ERR,"Error: socket() failed");
+		freeaddrinfo(results);
+		exit(1);
+	}
+
+
+	//3. Bind / assign name to a socket
+	printf("Binding the socket descriptor:%d.\n",sfd);
+	int bind_ret = bind(sfd, results->ai_addr, results->ai_addrlen);
+	if(bind_ret == -1){
+		printf("Error: Bind failed\n");
+		syslog(LOG_ERR,"Error: Bind failed");
+		printf("bind error: %s\n",strerror(errno));
+		freeaddrinfo(results);
+		exit(1);
+	}
+
+	//variables required for packet detection
+	int packet_size = 0;
+	char c = 0;
+
+	//Create file
+	int fd = creat(file_path, 0644);
+	if(fd == -1){
+		printf("Error: File could not be created!\n");
+		syslog(LOG_ERR,"Error: File could not be created!");
+		exit(1);
+	}
+
+	//close fd after creating
+	close(fd);
+
+	//free after use
+	freeaddrinfo(results);
+
+	while(1){
+
+		//make the buffer required for client input storage
+		op_buffer = (char *) malloc(sizeof(char)*BUFF_SIZE);
+		if(op_buffer == NULL){
+			printf("Malloc failed!\n");
+			exit(1);
+		}
+
+		memset(op_buffer,0,BUFF_SIZE);
+
+		//4. Listen on the socket
+		printf("Listening on socket.\n");
+		int list_ret = listen(sfd, LISTEN_BACKLOG);
+		if(list_ret == -1){
+			printf("Error: Listen failed\n");
+			syslog(LOG_ERR,"Error: Listen failed");
+			freeaddrinfo(results);
+			exit(1);
+		}
+
+		//5. accept the socket
+		client_addr_size = sizeof(struct sockaddr);
+
+		printf("Accepting connection.\n");
+		accept_fd = accept(sfd,(struct sockaddr *)&client_addr, &client_addr_size);
+		if(accept_fd == -1){
+			printf("Error: accept failed\n");
+			printf("accept error: %s\n",strerror(errno));
+			syslog(LOG_ERR,"Error: accept failed");
+			freeaddrinfo(results);
+			exit(1);
+		}
+
+		//Below logic required to get ip address in readable format.
+		struct sockaddr_in *addr = (struct sockaddr_in *)&client_addr;
+
+		inet_ntop(AF_INET, &(addr->sin_addr),ipv_4,INET_ADDRSTRLEN);
+
+		//Logging the client connection and address
+		syslog(LOG_DEBUG,"Accepting connection from %s",ipv_4);
+		printf("Accepting connection from %s\n",ipv_4);
+
+		//Package storage related variables
+		bool packet_comp = false;
+		int i;
+		int recv_ret = 0;
+
+		/*Packet reception, detection and storage logic*/
+		while(packet_comp == false){
+
+			//printf("Receiving data from descriptor:%d.\n",sfd);
+
+			recv_ret = recv(accept_fd, buff, BUFF_SIZE ,0); //**!check the flag
+			if(recv_ret < 0){
+				printf("Error: Receive failed\n");
+				printf("recv error: %s\n",strerror(errno));
+				syslog(LOG_ERR,"Error: Receive failed");
+				exit(1);
+			}else if(recv_ret == 0){
+				break;
+			}
+
+			/*Detect '\n' or ASCII value 
+			  10 in the packet.
+			*/
+			for(i = 0;i < BUFF_SIZE;i++){
+
+				if(buff[i] == 10){
+					packet_comp = true;
+					i++;
+					break;
+				}
+
+			}
+			packet_size += i;
+
+			/*reallocate to a larger buffer now as static buffer can
+			  only accomodate upto fixed size*/
+			op_buffer = (char *) realloc(op_buffer,(packet_size+1));
```

Comment:
@vishalraj3112 what assumption is made here about the size of RAM heap relative to the total size of all packets sent?